src/daft-core/src/series/ops/shift.rs

```diff
@@ -0,0 +1,56 @@
+use crate::datatypes::DataType;
+use crate::series::Series;
+use common_error::DaftError;
+use common_error::DaftResult;
+
+impl Series {
+    pub fn shift_left(&self, bits: &Self) -> DaftResult<Series> {
+        use crate::series::array_impl::IntoSeries;
+        if !bits.data_type().is_integer() {
+            return Err(DaftError::TypeError(format!(
+                "Expected input to shift_left to be integer, got {}",
+                bits.data_type()
+            )));
+        }
+        let bits = bits.cast_to_uint64()?;
```

Comment:
Instead of having a dedicated cast_to_uint64 function, you can do a check on the array to see if there are negative numbers, and then do the normal `bits.cast(&DataType::UInt64)?;`